### PR TITLE
feat: include support for metrics

### PR DESCRIPTION
**Pull request by acuarica**
**Description**:

This PR adds another endpoint, `/metrics`, where you can fetch service metrics to be ingested by Prometheus/Grafana.

Using for instance the [NodeJS Application Dashboard](https://grafana.com/grafana/dashboards/11159-nodejs-application-dashboard/) you can see NodeJS metrics

image

It also exports requests and events (response) triggered during verification, for example

```
sourcify_http_request_duration_seconds_bucket{le="0.003",status_code="200",method="POST",path="/verify"} 3
sourcify_http_request_duration_seconds_bucket{le="0.03",status_code="200",method="POST",path="/verify"} 4
sourcify_http_request_duration_seconds_bucket{le="0.1",status_code="200",method="POST",path="/verify"} 5
sourcify_http_request_duration_seconds_bucket{le="0.3",status_code="200",method="POST",path="/verify"} 5
sourcify_http_request_duration_seconds_bucket{le="1.5",status_code="200",method="POST",path="/verify"} 6
sourcify_http_request_duration_seconds_bucket{le="10",status_code="200",method="POST",path="/verify"} 6
sourcify_http_request_duration_seconds_bucket{le="+Inf",status_code="200",method="POST",path="/verify"} 6
sourcify_http_request_duration_seconds_sum{status_code="200",method="POST",path="/verify"} 0.632749876
sourcify_http_request_duration_seconds_count{status_code="200",method="POST",path="/verify"} 6

# HELP sourcify_up 1 = up, 0 = not up
# TYPE sourcify_up gauge
sourcify_up 1

# HELP sourcify_event_count events that happened during verification labeled with: event and chainId
# TYPE sourcify_event_count counter
sourcify_event_count{event="Verification.MatchStored",chainId="298"} 2
```

Note that the `sourcify_` prefix is used all metrics [to easily differentiate in Grafana](https://github.com/hashgraph/hedera-sourcify/pull/131#discussion_r1484780548).

The events exported are 

```js
  "Verification.MatchStored": [],
  "Server.SourcifyChains.Warn": [],
  // [...]
  "SourceFetcher.UsingFallback": [],
  "SourceFetcher.NewSubscription": [],
  "SourceFetcher.Cleanup": [],
  "SourceFetcher.FetchFailed": [],
  "SourceFetcher.FetchingSuccessful": [],
```

(the ones omitted `[...]` are `Monitor.*` related not used at the moment)

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: src/server/server.ts [hunk 1]
```diff
@@ -175,6 +181,33 @@ export class Server {
...
+      promClient: {
+        collectDefaultMetrics: {
+        }
+      }
+    });
+    // add the prometheus middleware to all routes
+    this.app.use(metricsMiddleware);
+
+    const counter = new client.Counter({
+      name: 'event_count',
```
Comment (Nana-EC) [inline src/server/server.ts:199]: Probably good to add a verification service related prefix so it's easy to differentiate in grafana

**Review by Nana-EC — COMMENTED** [review]
LG
Just naming suggestion

**Review by AlfredoG87 — APPROVED** [review]
LGTM, nice work!

I like that you used the express-prom-bundle and did not had to do things from scratch! 💯

File: src/server/server.ts (hunk 1 above)
Comment (AlfredoG87) [reply src/server/server.ts:199]: I agree on this request, is also possible to add a prefix using the prometheus configuration, but I agree that having a Const defined a the project level to name all other app related metrics is the recommended way, so is easier to identify the metrics regardless of the prometheus scrapping config that is being used.

Comment (acuarica) [reply src/server/server.ts]: Done. Also updated the PR description including this change.

**Review by svienot — APPROVED** [review]
Looks good to me.

**Pull request merged**